src/Main.cpp

```diff
@@ -26,14 +28,36 @@ int main()
             sf::Event event;
             while (window.pollEvent(event))
             {
-                if (event.type == sf::Event::Closed)
+              if (event.type == sf::Event::Closed)
                     window.close();
+               
+     
+            switch (event.type)
+            paused = !paused;
+            if (event.type == sf::Event::KeyReleased)
+            {
+                if (event.key.code == sf::Keyboard::Escape);
             }
+                //std::cout << "Esc was pressed" << std::endl;
+            {
+                if (event.type == sf::Event::KeyPressed)
+                paused = false;
+                else if (Keyboard::isKeyPressed(Keyboard::Escape))
+                paused = true;
+                if (event.type == sf::Event::KeyPressed)
 
+                   
+                break;
+               
+             }
+             
+            
+            }
+           
+            if (!paused) 
```

Comment:
That's one of the peculiarities of C++. Writing ```if``` statement without curly braces is allowed, but it means that ONLY the next statement will be enclosed in the statement. So what you have written here is equivalent to
```
if (!paused)
{
     scene.Update(dt);
}
scene.CheckCollisions();
scene.Draw();
```

If you want to enclose more statements in the ```if``` section, you have to use curly braces